src/components/ec-checkbox/ec-checkbox.spec.js

```diff
@@ -0,0 +1,192 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import EcCheckbox from './ec-checkbox.vue';
+
+function mountCheckbox(props, mountOpts) {
+  return mount(EcCheckbox, {
+    propsData: { ...props },
+    ...mountOpts,
+  });
+}
+
+function mountCheckboxAsTemplate(template, props, mountOpts) {
+  const localVue = createLocalVue();
+
+  const Component = localVue.extend({
+    components: { EcCheckbox },
+    template,
+  });
+
+  return mount(Component, {
+    localVue,
+    propsData: { ...props },
+    ...mountOpts,
+  });
+}
+
+describe('EcCheckbox', () => {
+  describe(':props', () => {
+    it(':label - should render the checkbox with a label', () => {
+      const wrapper = mountCheckbox({
+        label: 'Test label prop',
+      });
+
+      expect(wrapper.find('.ec-checkbox__label').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':errorMessage - should render the checkbox with an errorMessage', () => {
+      const wrapper = mountCheckbox({
+        errorMessage: 'Test error message prop',
+      });
+
+      expect(wrapper.find('.ec-checkbox__error-text').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':disabled - should render the checkbox with a disabled attribute', () => {
+      const wrapper = mountCheckbox({
+        disabled: true,
+      });
+
+      expect(wrapper.element).toMatchSnapshot();
+    });
+  });
+
+  describe('#slots', () => {
```

Comment:
We should test scenarios when you pass both, `label` prop and `label` slot. Slot should always win.

Same for error-message.